### PR TITLE
Fix Python3.11 tests

### PR DESCRIPTION
**Pull request by JanuszL**
- moves to a newer PaddlePaddle version that supports Python3.11
- moves to a newer OpenCV version that supports Python3.11
- fixes deprecated the usage of inspect.getargspec on nose

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- moves to a newer PaddlePaddle version that supports Python3.11
- moves to a newer OpenCV version that supports Python3.11
- fixes deprecated the usage of inspect.getargspec on nose

### Affected modules and functionalities:
- nose_wrapper
- setup_packages.py

### Key points relevant for the review:
- NA

### Tests:

- [x] Existing tests apply
  - TL1_tensorflow_plugin
  - TL0_FW_iterators
  - TL1_python-self-test-slow
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Review by awolant — APPROVED** [review]

File: dali/python/nvidia/dali/_autograph/operators/exceptions.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def assert_stmt(expression1, expression2):
   """
   if not callable(expression2):
     raise ValueError('{} must be a callable'.format(expression2))
-  args, _, keywords, _ = inspect.getargspec(expression2)
+  args, _, keywords, _, _, _, _ = inspect.getfullargspec(expression2)
```
Comment (mzient) [inline dali/python/nvidia/dali/_autograph/operators/exceptions.py:48]: Maybe:
```suggestion
  args, _, keywords, *_ = inspect.getfullargspec(expression2)
```
?

Comment (JanuszL) [reply dali/python/nvidia/dali/_autograph/operators/exceptions.py]: Done

File: qa/nose_wrapper/__main__.py [hunk 2]
```diff
@@ -6,14 +6,23 @@
...
 import nose.plugins.attrib
+import inspect
+
 if sys.version_info >= (3, 10) and not hasattr(collections, "Callable"):
     nose.case.collections = collections.abc
     nose.inspector.collections = collections.abc
     nose.loader.collections = collections.abc
     nose.suite.collections = collections.abc
     nose.plugins.attrib.collections = collections.abc
+if sys.version_info >= (3, 11):
```
Comment (mzient) [inline qa/nose_wrapper/__main__.py:17]: Total nitpick: some empty lines to separate those rather unrelated workarounds.

**Review by mzient — APPROVED** [review]

File: qa/nose_wrapper/__main__.py (hunk 2 above)
Comment (JanuszL) [reply qa/nose_wrapper/__main__.py:17]: Done

**Pull request merged**